doc/development/plugins.rst

```diff
@@ -180,23 +293,39 @@ PennyLane uses a ``setuptools`` ``entry_points`` approach to plugin discovery/in
+is the short name of the device, and ``MyModule.MySubModule`` is the path to your Device class,
+``MyDevice1``.
 
-To ensure your device is working as expected, you can install it in developer mode using ``pip install -e pluginpath``, where ``pluginpath`` is the location of the plugin. It will then be accessible via PennyLane.
+To ensure your device is working as expected, you can install it in developer mode using
+``pip install -e pluginpath``, where ``pluginpath`` is the location of the plugin. It will
+then be accessible via PennyLane.
 
 
 Testing
-^^^^^^^
+-------
 
-All plugins should come with extensive unit tests, to ensure that the device supports the correct gates and observables, and is applying them correctly. For an example of a plugin test suite, see ``tests/test_default_qubit.py`` and ``tests/test_default_gaussian.py`` in the main `PennyLane repository <https://github.com/XanaduAI/pennylane/>`_.
+All plugins should come with extensive unit tests, to ensure that the device supports the correct
+gates and observables, and is applying them correctly. For an example of a plugin test suite, see
+``tests/test_default_qubit.py`` and ``tests/test_default_gaussian.py`` in the main
+`PennyLane repository <https://github.com/XanaduAI/pennylane/>`_.
 
-In general, as all supported operations have their gradient formula defined and tested by PennyLane, testing that your device calculates the correct gradients is not required — just that it *applies* and *measures* quantum operations and observables correctly.
+Integration tests to check that the expectation values, variance, and samples are correct for
+various circuits and observables are provided in the
+`PennyLane Plugin Template <https://github.com/XanaduAI/pennylane-plugin-template>`__ repository.
+
+In general, as all supported operations have their gradient formula defined and tested by
+PennyLane, testing that your device calculates the correct gradients is not required — just
+that it *applies* and *measures* quantum operations and observables correctly.
 
 
 Supporting new operations
-^^^^^^^^^^^^^^^^^^^^^^^^^
+-------------------------
 
-If you would like to support an operation or observable that is not currently supported by PennyLane, you can subclass the :class:`~.Operation` and :class:`~.Observable` classes, and define the number of parameters the operation takes, and the number of wires the operation acts on. For example, to define the Ising gate :math:`XX_\phi` depending on parameter :math:`\phi`,
+If you would like to support an operation or observable that is not currently supported by
+PennyLane, you can subclass the :class:`~.Operation` and :class:`~.Observable` classes, and
+define the number of parameters the operation takes, and the number of wires the operation
+acts on. For example, to define the Ising gate :math:`XX_\phi` depending on parameter :math:`\phi`,
 
 .. code-block:: python
```

Comment:
I don't know about the Ising gate -- does it really have the default gradient recipe?

Reply:
Good point - I guess I'll need to go through the maths and see.

Maybe I should just change the name to `CustomOperation` :laughing:

Reply:
I think that is actually a good idea :+1: